packages/xchain-thorchain-amm/src/thorchain-amm.ts

```diff
@@ -309,6 +309,8 @@ export class ThorchainAMM {
    */
   public async isRouterApprovedToSpend({ asset, amount, address }: IsApprovedParams): Promise<string[]> {
     const errors: string[] = []
+    const isERC20 = isProtocolERC20Asset(asset)
+    if (!isERC20) errors.push('Asset should be ERC20')
```

Comment:
Replace the condition with `isProtocolERC20Asset(asset)` directly

Reply:
Ok, [fixed here](https://github.com/xchainjs/xchainjs-lib/pull/1243/commits/7f45944d02019f7eb1170d7f6ac3d9e7dea70e0a)